### PR TITLE
✨(security) add http basic auth for app services

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

In some cases, we must be able to protect a publicly deployed service behind an HTTP Basic Auth, _e.g._ to prevent search engines bots crawling or restrict access to a confidential application instance.

## Proposal

- [x] automate htpasswd file generation for apps
- [x] integrate HTTP Basic Auth for `richie`
- [x] integrate HTTP Basic Auth for `marsha`
- [x] integrate HTTP Basic Auth for `edxapp`

### CODE REVIEW

File: tasks/get_vault_for_app.yml [hunk 1]
```diff
@@ -19,3 +19,20 @@
...
+- name: Test htpasswd target file for app
+  stat:
+    path: "{{ app_htpasswd_file_path }}"
+  register: app_htpasswd_file
+  tags: htpasswd
+
+- name: Set application htpasswd variable
+  set_fact:
+    "{{ app.name }}_htpasswd": "{{ app_htpasswd_file.stat.path }}"
+  when: activate_http_basic_auth and app_htpasswd_file.stat.exists and app_htpasswd_file.stat.isreg
```
Comment (sampaccoud) [inline tasks/get_vault_for_app.yml:39]: can you comment what isreg is doing checking? here or in the previous task?

File: apps/richie/templates/nginx/secret.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app: richie
+    service: "nginx"
+  name: "{{ richie_nginx_htpasswd_secret_name }}"
+  namespace: "{{ project_name }}"
+data:
+  "{{ http_basic_auth_user_file | basename }}": "{{ lookup('file', richie_htpasswd) | b64encode }}"
```
Comment (sampaccoud) [inline apps/richie/templates/nginx/secret.yml.j2:13]: took me some time to understand wher the "richie_htpasswd" variable comes from (same for marsha and edxapp), maybe add a comment with link to the line where it's done in `get_vault_for_app`.

**Review by sampaccoud — APPROVED** [review]

**Comment by sampaccoud** [comment]
@jmaupetit awesome feature, and nicely done! Just added comments where I found it hard to understand if you want to add comments.

File: apps/richie/templates/nginx/secret.yml.j2 (hunk 2 above)
Comment (jmaupetit) [reply apps/richie/templates/nginx/secret.yml.j2:13]: Yeah, that's a tricky part. Maybe I should make it simpler.

**Pull request merged**